### PR TITLE
Add number-based pagination

### PR DESCRIPTION
**Pull request by gris**
Solves #116.

### CODE REVIEW

**Comment by arademaker** [comment]
@fcbr I will let you review it right ?

File: templates/search-result.tmpl [hunk 1]
```diff
@@ -136,12 +136,24 @@
...
       
-      {if $previous >= 0}
+    {if $previous >= 0}
 	<span class="txtPrev">
-	  <a href="#" onclick="javascript:jumpto({$previous}); return false;">&laquo; Previous</a>
+	  <a href="#" onclick="javascript:jumpto({$previous}); return false;">&laquo;&nbsp;Previous</a>
 	</span>
 	{/if}
-
+    {for $f in range(max(0, floor($previous/$limit) - 4),
```
Comment (fcbr) [inline templates/search-result.tmpl]: -4 and +5 in the next line are related to the step, no?  Notice that the step is configurable (see method `get-next` in `web-service.lisp`, default is 10).  We should probably pass the step as a parameter and use the parameter as opposed to hardcoding numbers in the template file.

**Review by fcbr — CHANGES_REQUESTED** [review]
Please make sure the configurable step is taken into account.  You might need to pass it explicitly as a parameter now.

File: templates/search-result.tmpl (hunk 1 above)
Comment (gris) [reply templates/search-result.tmpl]: The hard coded numbers -4 and 5 are related to the number of Pages that is shown to the user and not to the step. I made This decision to show always 5 pages ahead and before the page that is already displayed

**Comment by arademaker** [comment]
This decision must go in the configuration , not harder code in the template .

File: templates/search-result.tmpl (hunk 1 above)
Comment (fcbr) [reply templates/search-result.tmpl]: Ah, OK.  Please make sure these are also parameterized; it's best to avoid these hardcoded constants.

**Comment by gris** [comment]
I think that now all the problems are solved.

**Review by fcbr — APPROVED** [review]

**Pull request merged**